scripts/hid_configurator/modules/dfu.py

```diff
@@ -366,117 +415,142 @@ def file_crc(dfu_image):
     return crc32
 
 
-def dfu_sync_wait(dev, is_active):
-    if is_active:
-        dfu_state = 0x01
-    else:
-        dfu_state = 0x00
+def dfu_sync_wait_until_inactive(dev):
+    state_shown = False
 
-    for _ in range(DFU_SYNC_RETRIES):
+    while True:
         dfu_info = dfu_sync(dev)
+        if dfu_info is None:
+            break
 
-        if dfu_info is not None:
-            if dfu_info[0] != dfu_state:
-                # DFU may be transiting its state. This can happen when previous
-                # interrupted DFU operation is timing out. Sleep to allow it
-                # to settle the state.
-                time.sleep(DFU_SYNC_INTERVAL)
-            else:
-                break
+        if dfu_info.is_cleaning() and not state_shown:
+            print('Waiting for device to clean update slot')
+            state_shown = True
+
+        if dfu_info.is_busy():
+            # DFU may be transiting its state. This can happen when previous
+            # interrupted DFU operation is timing out. Sleep to allow it
+            # to settle the state.
+            time.sleep(DFU_SYNC_INTERVAL)
+        else:
+            break
 
     return dfu_info
 
 
 def dfu_transfer(dev, dfu_image, progress_callback):
     img_length = os.stat(dfu_image).st_size
-    dfu_info = dfu_sync_wait(dev, False)
-
-    if is_dfu_operation_pending(dfu_info):
-        return False
-
     img_csum = file_crc(dfu_image)
-
     if not img_csum:
         return False
 
-    offset = get_dfu_operation_offset(dfu_image, dfu_info, img_csum)
+    while True:
```

Comment:
This handles a case when you want to start a dfu with another image while previous image's data is still in RAM. Device will handle it well, that is it will start cleaning and reject the dfu. But script has no way of telling that as dfu start returns no info. I.e. You either get the device cleaning prior or post dfu start.

Reply:
Ok, we may make the script waiting for the flash area cleaning instead of informing about it and stopping execution.